File: program67.cpp
//Accept N nos and one other number and search that no.


#include<iostream>
using namespace std;

class ArrayX
{
    private:
        int *Arr;
        int iSize;

    public: 
        ArrayX(int iValue)
        {
            this->iSize = iValue;
            Arr = new int[iSize];
        }
        ~ArrayX()
        {
            delete []Arr;
        }
        void Accept()
        {
            int iCnt = 0;
            cout<<"Enter array elements: "<<endl;
            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                cin>>Arr[iCnt];
            }
        }
        void Display()
        {
            int iCnt = 0;
            cout<<"Array elements are : "<<endl;
            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                cout<<Arr[iCnt]<<endl;
            }
        }
        bool Search()
        {
            int iCnt = 0, iNo = 0;
            bool flag = false;
            cout<<"Enter element to search"<<endl;
            cin>>iNo;

            for(iCnt = 0; iCnt < iSize; iCnt++)
            {
                if(Arr[iCnt] == iNo)
                {
                    flag = true;
                    break;
                }
            }
            return flag;
        }
};

int main()
{
    bool bRet = 0;
    ArrayX obj(5);

    obj.Accept();
    obj.Display();
    bRet = obj.Search();
    if(bRet == true)
    {
        cout<<"Element is present there."<<endl;
    }
    else{
        cout<<"Element is not present there."<<endl;
    }
    
    return 0;
}

File: program67.java
//Accept N nos and one other number and search that no.
import java.lang.*;
import java.util.*;

class ArrayX
{
    private int Arr[];

    public ArrayX(int iNo)
    {
        Arr = new int[iNo];
    }

    public void Accept()
    {
        int iCnt = 0;
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter array elements:");
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        int iCnt = 0;
        System.out.println("Array elements are:");
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]); 
        }
    }

    public boolean Search()
    {
        int iCnt = 0, iNum = 0 ;
        boolean flag = false;
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements to search:");
        iNum = sobj.nextInt();

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if( Arr[iCnt] == iNum)
            {
                flag = true;
                break;
            }
        }
        return flag;
    }
}

class program67
{
    public static void main(String arg[])
    {
        boolean bRet;
        int iLength = 0;

        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter value for constructor : ");
        iLength = sobj.nextInt();

        ArrayX obj = new ArrayX(iLength);

        obj.Accept();
        obj.Display();

        bRet = obj.Search();

        if(bRet == true)
        {
            System.out.println("Element is there in array");
        }
        else
        {
            System.out.println("Element is not there in array");
        }
    }
}